### ISSUE TITLE
Function may not auto-compile after udpating csx file

### ISSUE THREAD

**Issue by fashaikh** [Azure/azure-webjobs-sdk#1065]
After saving function code via portal, logstream still shows previous code getting executed
I added a couple of log.Info lines in there, and hit save (and Save and Run) .  The PUT request shows 204 Accepted, and I can see the update via Kudu, but seems the new code doesn't get compiled.  Newer log stream output still reflects the old code (without log.info).

#### Repro steps

Provide the steps required to reproduce the problem

1. Step A May not be easy to repro. I have a functionapp with a few functions. This is the one that reproes (Consumption Plan)

`FunctionAppName: appserviceemailer
Function: Test`

2. Step B

#### Expected behavior

The updated code gets picked up and compiled and is available in the logstream output

#### Actual behavior

The logstream output of the execution still shows the old code getting executed. There is also no `Function compilation succeeded/failed` 
message.

#### Known workarounds

Not sure. Will restart the site. But this scenario should work smoothly without having to restart. 
###Update : This worked after restarting the site . 

#### Related information 

Provide any related information 

* Package version : Latest
* Links to source : Provided above

**Comment by christopheranderson** [comment]
Could you please give us an approximate UTC timestamp for this?

**Comment by fashaikh** [comment]
Not sure. It was just about the time I opened this bug. 
Looking at this 
https://api.github.com/repos/Azure/azure-webjobs-sdk/issues/1065 
About an hour range before ` 2017-03-15T22:27:56Z `

**Comment by christopheranderson** [comment]
Do you see this over and over again, or is it just one in a while?

**Comment by fashaikh** [comment]
Seen it at least thrice. Doesn't happen always though. Not sure what triggers it go in that state.
Obviously I have to be using the portal rigorously for it to happen. I haven't touched the code there in a while.
I believe Ben also faced this issue with his runners/alerts function app.

**Comment by mathewc** [comment]
Most likely explanation is issues with our file watch. @fabiocav has been doing work to address this, see https://github.com/Azure/azure-webjobs-sdk-script/issues/1277. Please reopen if the issue repros after those changes go in.

**Comment by fashaikh** [comment]
Its still happening.  Let me know if you need mroe details
![image](https://user-images.githubusercontent.com/2650941/28739185-a0c8f834-73ad-11e7-8798-eacf2b878bc4.png)

![image](https://user-images.githubusercontent.com/2650941/28739191-ae38bd38-73ad-11e7-8b79-e5299ad04740.png)

**Comment by fashaikh** [comment]
@fabiocav  After updating the project.json file . the function executes and I see the email it generates, but the console window still says

`
2017-07-29T00:05:08  No new trace in the past 7 min(s).
2017-07-29T00:06:08  No new trace in the past 8 min(s).
2017-07-29T00:07:08  No new trace in the past 9 min(s).
2017-07-29T00:08:08  No new trace in the past 10 min(s).
2017-07-29T00:09:08  No new trace in the past 11 min(s). 

`

**Comment by fashaikh** [comment]
I'll go ahead and restart the functionapp for now

**Comment by christopheranderson** [comment]
Assigning to on-call ( @davidebbo ) to see what's going on now to confirm it's file issue

**Comment by davidebbo** [comment]
When you're in that state, if you go to `D:\home\LogFiles\Application\Functions\function\YourFunc`, do you see new files appearing as things run?

Also, under `D:\home\LogFiles\Application\Functions\Host`, do you have a file called `debug_sentinel`?

**Comment by fashaikh** [comment]
Yes there is a `debug_sentinel` file present .  I can check the  `D:\home\LogFiles\Application\Functions\function\YourFunc` location the next time this happens.
Right now there are just two files in most active functions directories with the same approximate timestamps as below
![image](https://user-images.githubusercontent.com/2650941/28803321-888b13a0-7610-11e7-93cf-93d9ae83aedc.png)

**Comment by davidebbo** [comment]
Can someone with repo access please unassign me from this issue? Apparently, I can't do it myself, and it's polluting my https://github.com/issues/assigned list. 😓 Thanks!
